pennylane/wires.py

```diff
@@ -0,0 +1,322 @@
+            # If input is an iterable, interpret as a collection of wire indices
+            self.wire_list = _clean(wires)
+
+        elif isinstance(wires, int) and wires >= 0:
+            # If input is non-negative integer, interpret as single wire index
+            # and wrap as a list
+            self.wire_list = [wires]
+
+        else:
+            raise WireError(
+                "received unexpected wires input {} of type {}.".format(wires, type(wires))
+            )
+
+    def __getitem__(self, idx):
+        """Method to support indexing."""
+        return self.wire_list[idx]
+
+    def __len__(self):
+        """Method to support ``len()``."""
+        return len(self.wire_list)
+
+    def __repr__(self):
+        """Method defining the string representation of this class."""
+        return "<Wires = {}>".format(self.wire_list)
+
+    def __eq__(self, other):
+        """Method to support the '==' operator."""
+        if isinstance(other, self.__class__):
+            return self.wire_list == other.wire_list
+        else:
+            return False
+
+    def __ne__(self, other):
+        """Method to support the '!=' operator."""
+        return not self.__eq__(other)
+
+    def combine(self, wires):
+        """Return a copy of this ``Wires`` object extended by the unique wires of ``wires``.
+
+        For example:
+
+        >>> wires1 =  Wires([4, 0, 1])
+        >>> wires2 = Wires([3, 0, 4])
+        >>> wires1.combine(wires2)
+        <Wires [4, 0, 1, 3]>
+
+        Args:
+            wires (Wires): A Wires object whose unique indices are combined with this Wires object.
+
+        Returns:
+            Wires: combined wires
+        """
+
+        if not isinstance(wires, Wires):
+            raise WireError(
+                "expected a `pennylane.wires.Wires` object; got {} of type {}".format(
+                    wires, type(wires)
+                )
+            )
+
+        combined_wire_list = deepcopy(self.wire_list)
+        combined_wire_list.extend(wires.wire_list)
+        return Wires(combined_wire_list)
+
+    def intersect(self, wires):
+        """
+        Creates a wires object that is the intersection of wires between 'wires' and this object.
+
+        For example:
+
+        >>> wires1 =  Wires([4, 0, 1])
+        >>> wires2 = Wires([0, 4, 3])
+        >>> wires1.intersect(wires2)
+        <Wires = [4, 0]>
+
+        The returned wire sequence is ordered according to the order of this Wires object.
+
+        Args:
+            wires (Wires): A Wires class object
+
+        Returns:
+            Wires: intersection of this Wires object and 'wires'
+        """
+
+        if not isinstance(wires, Wires):
+            raise WireError(
+                "expected a `pennylane.wires.Wires` object; got {} of type {}.".format(
+                    wires, type(wires)
+                )
+            )
+
+        intersect = [w for w in self.wire_list if w in wires.wire_list]
```

Comment:
I suppose the order of the intersect matters as well? e.g., `wires2.intersect(wires1) != wires1.intersect(wires2)`.

Reply:
Yes, since these are "set" operations for ordered sequences...to be honest I find them still a bit clumsy.

Reply:
Yeah, I almost left a comment along the lines of 'can't we just use the built in `set()`' until I realised the order mattered 🤔 Which is odd, still wrapping my head around it. Could it lead to odd side effects?

Reply:
I will carefully check the use cases.